### ISSUE TITLE
Embedding Model in YAML Not Updated After Changing Knowledge Context in Dify

### ISSUE THREAD

**Issue by kikumoto** [langgenius/dify#8306]
### Self Checks

- [X] This is only for bug report, if you would like to ask a question, please head to [Discussions](https://github.com/langgenius/dify/discussions/categories/general).
- [X] I have searched for existing issues [search for existing issues](https://github.com/langgenius/dify/issues), including closed ones.
- [X] I confirm that I am using English to submit this report (我已阅读并同意 [Language Policy](https://github.com/langgenius/dify/issues/1542)).
- [X] [FOR CHINESE USERS] 请务必使用英文提交 Issue，否则会被关闭。谢谢！:）
- [X] Please do not modify this template :) and fill in all the required fields.

### Dify version

0.8.0

### Cloud or Self Hosted

Self Hosted (Docker)

### Steps to reproduce

1. Create two Knowledges in Dify:
   a. Specify Bedrock's `cohere.embed-multilingual-v3` as the Embedding Model (Knowledge A)
   b. Specify Azure OpenAI Service's `text-embedding-3-large` as the Embedding Model (Knowledge B)

2. In Studio, create a Chatbot - Basic:
   a. Add only Knowledge A in the Context
   b. Publish the chatbot
   c. Export DSL

At this stage, the YAML file contains the following settings:

```
model_config.dataset_configs.reranking_mode: weighted_score
model_config.dataset_configs.weights:vector_setting.embedding_model_name: cohere.embed-multilingual-v3
model_config.dataset_configs.weights:vector_setting.embedding_provider_name: bedrock
```

3. Next:
   a. Remove Knowledge A from the Context
   b. Add Knowledge B to the Context
   c. Publish the chatbot
   d. Export DSL again
At this stage, the YAML file remains unchanged:

```
model_config.dataset_configs.reranking_mode: weighted_score
model_config.dataset_configs.weights:vector_setting.embedding_model_name: cohere.embed-multilingual-v3
model_config.dataset_configs.weights:vector_setting.embedding_provider_name: bedrock
```

Additionally, the UI does not provide a way to edit the embedding_model_name in the `Retrieval Setting`.

### ✔️ Expected Behavior

This behavior is problematic when switching between different Embedding Models, a common practice when experimenting. The expected outcome should be:

```
model_config.dataset_configs.reranking_mode: weighted_score
model_config.dataset_configs.weights:vector_setting.embedding_model_name: text-embedding-3-large
model_config.dataset_configs.weights:vector_setting.embedding_provider_name: azure_openai
```

**Comment by crazywoola** [comment]
This is fixed in #8183)

**Comment by kikumoto** [comment]
@dosu 
We are not encountering a rate limit issue.
Also, waiting does not resolve the problem.

Could there be an issue with the Payload when calling the api/apps/xxxxxx/mode-config API?
In the Payload,

dataset_configs -> reranking_mode
and
dataset_configs -> weights -> vector_setting -> embedding_model_name
have not been changed from the values initially registered for the knowledge base.
As a result, I suspect that the database side is not being updated.

Could you please review the situation once again?
